web/crux-ui/src/models/container.ts

```diff
@@ -36,191 +38,302 @@ export type PortRange = {
   to: number
 }
 
-export type ExplicitContainerConfigPortRange = {
+export type ContainerConfigPortRange = {
+  id: string
   internal: PortRange
   external: PortRange
 }
 
-export const EXPLICIT_CONTAINER_NETWORK_MODE_VALUES = ['none', 'host', 'bridge'] as const
-export type ExplicitContainerNetworkMode = typeof EXPLICIT_CONTAINER_NETWORK_MODE_VALUES[number]
+export type JsonKeyValue = { [key: string]: string }
 
-export const EXPLICIT_CONTAINER_RESTART_POLICY_TYPE_VALUES = [
-  '',
-  'always',
-  'unless_stopped',
-  'no',
-  'on_failure',
-] as const
-export type ExplicitContainerRestartPolicyType = typeof EXPLICIT_CONTAINER_RESTART_POLICY_TYPE_VALUES[number]
+export const CONTAINER_NETWORK_MODE_VALUES = ['none', 'host', 'bridge', 'overlay', 'ipvlan', 'macvlan'] as const
+export type ContainerNetworkMode = typeof CONTAINER_NETWORK_MODE_VALUES[number]
 
-export const EXPLICIT_CONTAINER_DEPLOYMENT_STRATEGY_VALUES = ['recreate', 'rolling'] as const
-export type ExplicitContainerDeploymentStrategyType = typeof EXPLICIT_CONTAINER_DEPLOYMENT_STRATEGY_VALUES[number]
+export const CONTAINER_RESTART_POLICY_TYPE_VALUES = ['undefined', 'always', 'unlessStopped', 'no', 'onFailure'] as const
+export type ContainerRestartPolicyType = typeof CONTAINER_RESTART_POLICY_TYPE_VALUES[number]
 
-export type ExplicitContainerConfigExpose = {
-  public: boolean
-  tls: boolean
-}
+export const CONTAINER_DEPLOYMENT_STRATEGY_VALUES = ['unknown', 'recreate', 'rolling'] as const
+export type ContainerDeploymentStrategyType = typeof CONTAINER_DEPLOYMENT_STRATEGY_VALUES[number]
+
+export const CONTAINER_EXPOSE_STRATEGY_VALUES = ['none', 'expose', 'expose_with_tls'] as const
+export type ContainerConfigExposeStrategy = typeof CONTAINER_EXPOSE_STRATEGY_VALUES[number]
+
+export const CONTAINER_VOLUME_TYPE_VALUES = ['ro', 'rw', 'rwx', 'mem', 'tmp'] as const
+export type VolumeType = typeof CONTAINER_VOLUME_TYPE_VALUES[number]
 
-export type ExplicitContainerConfigIngress = {
+export type ContainerConfigIngress = {
   name: string
   host: string
   uploadLimitInBytes?: string
 }
 
-export type ExplicitContainerConfigVolume = {
+export type ContainerConfigVolume = {
+  id: string
   name: string
   path: string
   size?: string
-  type?: string
+  type?: VolumeType
   class?: string
 }
 
-export type ExplicitContainerConfigLog = {
-  type: string
-  config: { [key: string]: string }
+export const CONTAINER_LOG_DRIVER_VALUES = [
+  'none',
+  'gcplogs',
+  'local',
+  'json-file',
+  'syslog',
+  'journald',
+  'gelf',
+  'fluentd',
+  'awslogs',
+  'splunk',
+  'etwlogs',
+  'logentries',
+] as const
```

Comment:
Missing new line. Have you checked the code with the linter?

Reply:
I think its a github bug. There is a new line.